src/ape_ethereum/provider.py

```diff
@@ -592,7 +592,9 @@ def estimate_gas_cost(self, txn: TransactionAPI, block_id: Optional["BlockID"] =
     @cached_property
     def chain_id(self) -> int:
         default_chain_id = None
-        if (not self.network.is_adhoc and self.network.is_custom) or not self.network.is_dev:
+        if (not self.network.is_adhoc and self.network.is_custom) or (
+            not self.network.is_dev and not self.network.is_adhoc
+        ):
             # If using a live network, the chain ID is hardcoded.
             default_chain_id = self.network.chain_id
```

Comment:
I wanted to break these two conditions up into 2 so it reads easier. This is okay because `default_chain_id` is set to `None` by default from not matching any condition

```suggestion
        if not self.network.is_adhoc and self.network.is_custom:
            # If using a configured, custom network, the chain ID is hardcoded.
            default_chain_id = self.network.chain_id

        elif not self.network.is_dev and not self.network.is_adhoc:
            # If using a configured public network, the chain ID is hardcoded.
            default_chain_id = self.network.chain_id
```